File: ProjectPos.BackEnd/ProjectPos.Services/DTOs/PurchaceOrderLineDto.cs
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using ProjectPos.Services.EntityDtos;
using ProjectPos.Data.Shared.Enums;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace ProjectPos.Services.DTOs
{
    public class PurchaceOrderLineDto : EntityDto<int>
    {
        public int? OrderNumber { get; set; }
        public int? ProductInventoryId { get; set; }
        public string? Name { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? Quantity { get; set; }
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public Unit? Unit { get; set; }

        [Column(TypeName = "double(12, 2)")]
        public double? UnitPrice { get; set; }

        [Column(TypeName = "double(12, 2)")]
        public double? Price { get; set; }

        //foreign Key

        [ForeignKey("ProductInventoryId")]
        public ProductInventoryDto? Product { get; set; }
    }
}

File: ProjectPos.BackOffice.FrontEnd/src/proxy/interfaces/purchace-order-line-dto.ts
import { EntityDto } from "../entity-dtos/entity-dto";
import { Unit } from "../enums/unit";
import { ProductDto } from "./product-dto";
import { ProductInventoryDto } from "./product-inventory-dto";
import { PurchaceOrderDto } from "./purchace-order-dto";

export interface PurchaceOrderLineDto extends EntityDto<number> {
    orderNumber?: number;
    productInventoryId?: number;
    name?: string;
    quantity?: number;
    unit?: Unit;
    unitPrice?: number;
    price?: number;
    purchaceOrder?: PurchaceOrderDto;
    product?: ProductInventoryDto;
}
